src/app/shared/services/app-update/app-update.service.ts

```diff
@@ -0,0 +1,39 @@
+import { Injectable } from "@angular/core";
+import { AppUpdate, AppUpdateInfo, AppUpdateAvailability } from "@capawesome/capacitor-app-update";
+
+@Injectable({
+  providedIn: "root",
+})
+export class AppUpdateService {
+  constructor() {}
+
+  async checkForUpdatesAndAttemptImmediateUpdate() {
+    try {
+      const result = await AppUpdate.getAppUpdateInfo();
+      if (result.updateAvailability !== AppUpdateAvailability.UPDATE_AVAILABLE) {
+        return;
+      }
+      if (result.immediateUpdateAllowed) {
+        return await AppUpdate.performImmediateUpdate();
+      }
+    } catch (error) {
+      console.error("[App Update]", error);
+    }
+  }
+
+  async getAppUpdateInfo() {
+    return (await AppUpdate.getAppUpdateInfo()) as AppUpdateInfo;
+  }
+
+  async performImmediateUpdate() {
```

Comment:
nit(non-blocking)
If we're not planning on accessing these methods from other services probably no need to duplicate all the methods as public. If we move all init/handling logic to this service then probably can just rely on one or two functions